easybuild/easyblocks/b/boost.py

```diff
@@ -139,20 +142,33 @@ def configure_step(self):
         tup = (self.cfg['preconfigopts'], toolset, self.objdir, self.cfg['configopts'])
         run_cmd(cmd % tup, log_all=True, simple=True)
 
+        # Use build_toolset if specified or the bootstrap toolset without the OS suffix
+        self.toolset = self.cfg['build_toolset'] or re.sub('-linux$', '', toolset)
+
+        user_config = []
+
+        # Explicitely set the compiler path to avoid B2 checking some standard paths like /opt
+        cxx = os.getenv('CXX')
+        if cxx:
+            cxx = which(cxx, on_error=ERROR)
+            # Remove default toolset config which may lead to duplicate toolsets (e.g. for intel-linux)
+            apply_regex_substitutions('project-config.jam', [('using %s ;' % toolset, '')])
+            # Add our toolset config with no version and full path to compiler
+            user_config.append("using %s : : %s ;" % (self.toolset, cxx))
+
         if self.cfg['boost_mpi']:
 
             self.toolchain.options['usempi'] = True
             # configure the boost mpi module
             # http://www.boost.org/doc/libs/1_47_0/doc/html/mpi/getting_started.html
             # let Boost.Build know to look here for the config file
 
-            txt = ''
             # Check if using a Cray toolchain and configure MPI accordingly
             if self.toolchain.toolchain_family() == toolchain.CRAYPE:
                 if self.toolchain.PRGENV_MODULE_NAME_SUFFIX == 'gnu':
                     craympichdir = os.getenv('CRAY_MPICH2_DIR')
                     craygccversion = os.getenv('GCC_VERSION')
-                    txt = '\n'.join([
+                    user_config.extend([
                         'local CRAY_MPICH2_DIR =  %s ;' % craympichdir,
                         'using gcc ',
```

Comment:
Doesn't this conflict with the toolset setting above? i.e. the user_config.append...

Reply:
I'm not sure, maybe the last one wins... But added code to remove it and also check that it is actually used